src/catalog.c

```diff
@@ -224,74 +214,92 @@ catalog_owner(void)
 	return owner_oid;
 }
 
-Catalog *
-catalog_get(void)
+static const char *
+catalog_table_name(CatalogTable table)
 {
-	int			i;
-
-	if (!OidIsValid(MyDatabaseId))
-		elog(ERROR, "invalid database ID");
-
-	if (MyDatabaseId == catalog.database_id)
-		return &catalog;
+	return catalog_table_names[table].table_name;
+}
 
-	if (!extension_is_loaded() || !IsTransactionState())
-		return &catalog;
+/*
+ * Note that the first two arguments are pointers to populate.
+ * The rest of the arguments are used to populate the first two args.
+ */
+void
+catalog_populate_base(CatalogBase *catalog, CatalogTableInfo *tables, int max_tables, const TableInfoDef *table_ary, const TableIndexDef *index_ary, const char **serial_id_ary)
```

Comment:
naming convention for this is `catalog_base_init`

Reply:
also can we sparate out the thing that populates catalog base and the thing that populates the catalog table info?

Reply:
also isn't `tables` also an array? why doesn't it have the `ary` suffix if we are using that convention here

Reply:
Fixed rename and separated `catalog_database_info_init` from `catalog_tables_init`.

`tables` is an array, but it's not an input. Also renamed to `tables_info`.